Dark Mode issues Fixes

memoarfaa opened the pull request:
Fix Dark Mode issues 

Fixes #12027
Fixes #12992
Fixes #12991
Fixes #12582
Fixes #11941
Fixes #11932

memoarfaa commented:
> Hey @memoarfaa,
>
Hey @KlausLoeffelmann 

> thanks for this PR, but let's try to coordinate the efforts here, so we're not doing redundant work. Can you explain, which parts of the your PR is addressing which Issues, and point out the background a bit?
> 
Ok I will do

> We'll be addressing for Preview 5 (maybe early 6) the PRs which I did, since they had been planned and championed. We only have Monday 19th to complete testing, reviews and merge them, so the Preview 5 ship for any pivoting in approaches has sailed.

I understand that for Preview 5, only the already planned/championed PRs can be considered, and my changes won’t be merged at this stage.

> For Preview 6, I'd love to re-discuss approaches, though, so, let's keep the discussion rolling for improvements for the Preview 6 time frame!
>
For Preview 6, I’m happy to continue iterating on these improvements and would appreciate any feedback or direction from the team on priorities or preferred approaches.

> PS: If you have a good, feasible idea or approach, which we could take on (really!) short notice for the background-coloring of a read-only TextBox or RTF TextBox in DarkMode, without changing the default color (which would be too invasive), that would be really cool!!
>
About Read-Only TextBox Background Coloring in Dark Mode

I don't understand what you mean by 
>"without changing the default color"
>
If you mean the Win32 default theme Color the answer is no because it handled by WM_CTLCOLOR XXXX
if you mean the WinForms repo defaults BackColor and ForColor I'm using the same approach in Win32 WM_CTLCOLOR XXXX
explanation, 
 in TextBoxBase class
 ```
internal override HBRUSH InitializeDCForWmCtlColor(HDC dc, MessageId msg)
    {
#pragma warning disable WFO5001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
        bool isReadOnly = (PInvokeCore.GetWindowLong(this, WINDOW_LONG_PTR_INDEX.GWL_STYLE) & PInvoke.ES_READONLY) != 0;
        if (Application.IsDarkModeEnabled && isReadOnly)
        {
            switch (msg)
            {
                // In dark mode, a disabled edit control will paint the background and forground with light colors
                case PInvokeCore.WM_CTLCOLOREDIT:
                    PInvokeCore.SetBkColor(dc, 0x1A1A1A);
                    PInvokeCore.SetTextColor(dc, 0xffffff);
                    return PInvokeCore.CreateSolidBrush(0x1A1A1A);
                case PInvokeCore.WM_CTLCOLORSTATIC:
                    PInvokeCore.SetBkMode(dc, BACKGROUND_MODE.TRANSPARENT);
                    PInvokeCore.SetTextColor(dc, 0x6A6A6A);
                    return PInvokeCore.CreateSolidBrush(0x232323);
                default:
                    return base.InitializeDCForWmCtlColor(dc, msg);
            }
        }

#pragma warning restore WFO5001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
     else
        {
            if (msg == PInvokeCore.WM_CTLCOLORSTATIC && !ShouldSerializeBackColor())
            {
                // Let the Win32 Edit control handle background colors itself.
                // This is necessary because a disabled edit control will display a different
                // BackColor than when enabled.
                return default;

            }
            else
            {
                return base.InitializeDCForWmCtlColor(dc, msg);
            }
        }
    }

 ```

in RichTextBox Class
``` 
    private bool InternalSetForeColor(Color value)
    {

        CHARFORMAT2W cf = GetCharFormat(false);
        if ((cf.dwMask & CFM_MASK.CFM_COLOR) != 0
            && ColorTranslator.ToWin32(value) == cf.crTextColor)
        {
            return true;
        }

        cf.dwMask = CFM_MASK.CFM_COLOR;
        cf.dwEffects = 0;
#pragma warning disable WFO5001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
        cf.crTextColor = Application.IsDarkModeEnabled && ReadOnly
            ? 0x6A6A6A
            : ColorTranslator.ToWin32(value);
#pragma warning restore WFO5001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
        return SetCharFormat(PInvoke.SCF_ALL, cf);
    }

```

memoarfaa commented:
@Olina-Zhang 
If you have some time, can you confirm that all the issues in these Pull request have been resolved?

Olina-Zhang commented:
Hi @memoarfaa, tested this PR change to verify the GH issues mentioned in this PR, all are fixed besides GH issue: [#12027](https://github.com/dotnet/winforms/issues/12027), in addition, found 2 new issues:

- New issues:
![image](https://github.com/user-attachments/assets/9d58bcdd-a2ad-48af-a7ce-76a619cf9fd6)
![image](https://github.com/user-attachments/assets/fed6a766-f935-443e-ac25-c4cfa0332b5b)


- GH issues verification

1. [#12027](https://github.com/dotnet/winforms/issues/12027): Not fixed well, dropdown arrows on ToolStripSplitButton have the correct color, but ToolStripSeparator has wrong color, it's hard to visible
![image](https://github.com/user-attachments/assets/ece57e58-9cac-458b-bf73-7563aeaa0bfb)
2. [#12992](https://github.com/dotnet/winforms/issues/12992): fixed
3. [#12991](https://github.com/dotnet/winforms/issues/12991): fixed
4. [#12582](https://github.com/dotnet/winforms/issues/12582): fixed
5. [#11941](https://github.com/dotnet/winforms/issues/11941): fixed
6. [#11932](https://github.com/dotnet/winforms/issues/11932): fixed
7. [#13636](https://github.com/dotnet/winforms/issues/13636): fixed

Any question, please let me know.

memoarfaa commented:
> Hi @memoarfaa, tested this PR change to verify the GH issues mentioned in this PR, all are fixed besides GH issue: [#12027](https://github.com/dotnet/winforms/issues/12027), in addition, found 2 new issues:
> 
> * New issues:
>   ![image](https://private-user-images.githubusercontent.com/26474449/457826456-9d58bcdd-a2ad-48af-a7ce-76a619cf9fd6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA4ODk0MTAsIm5iZiI6MTc1MDg4OTExMCwicGF0aCI6Ii8yNjQ3NDQ0OS80NTc4MjY0NTYtOWQ1OGJjZGQtYTJhZC00OGFmLWE3Y2UtNzZhNjE5Y2Y5ZmQ2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjI1VDIyMDUxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMxM2M0NzY3ZjBkNjgwYmM4MGExZTVjOWYwZTIxMTM5OGVjOGM3NmJjZjNiMTFkMWQxMjAyNDUxYTNhYWM2Y2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.q8gwR4oevkkSmihLGOo-gB-Hd5hEmoApDn7ahoEZfLs)
>   ![image](https://private-user-images.githubusercontent.com/26474449/457826623-fed6a766-f935-443e-ac25-c4cfa0332b5b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA4ODk0MTAsIm5iZiI6MTc1MDg4OTExMCwicGF0aCI6Ii8yNjQ3NDQ0OS80NTc4MjY2MjMtZmVkNmE3NjYtZjkzNS00NDNlLWFjMjUtYzRjZmEwMzMyYjViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjI1VDIyMDUxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE3NjZiMDIxYmI5NzhmN2UyN2NmNmRmZWEwZDgxMzBhNmU2OTc2Zjk0Mjc5ZGVmODUyMTcwZTYwNDU4MWUyZmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.RbYq5uc-O1nkCl4g7KmYp5fEJUQBk6GoDTpBu7uXwhA)
> * GH issues verification
> 
> 1. [#12027](https://github.com/dotnet/winforms/issues/12027): Not fixed well, dropdown arrows on ToolStripSplitButton have the correct color, but ToolStripSeparator has wrong color, it's hard to visible
>    ![image](https://private-user-images.githubusercontent.com/26474449/457825571-ece57e58-9cac-458b-bf73-7563aeaa0bfb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA4ODk0MTAsIm5iZiI6MTc1MDg4OTExMCwicGF0aCI6Ii8yNjQ3NDQ0OS80NTc4MjU1NzEtZWNlNTdlNTgtOWNhYy00NThiLWJmNzMtNzU2M2FlYWEwYmZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjI1VDIyMDUxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBmODllOTI2YjQxNDQ5YjNhZWMwM2E0Zjk5ZmVhZDQwZjU3ZDM2MTkyYjg0MmQzODYyZDJkMGQ3OWU5ODk1MzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.b56yfgNeWS_Qu58qHPNfT86P7YmPQh5Gt3VKoWqvYGc)
> 2. [#12992](https://github.com/dotnet/winforms/issues/12992): fixed
> 3. [#12991](https://github.com/dotnet/winforms/issues/12991): fixed
> 4. [#12582](https://github.com/dotnet/winforms/issues/12582): fixed
> 5. [#11941](https://github.com/dotnet/winforms/issues/11941): fixed
> 6. [#11932](https://github.com/dotnet/winforms/issues/11932): fixed
> 7. [#13636](https://github.com/dotnet/winforms/issues/13636): fixed
> 
> Any question, please let me know.

@KlausLoeffelmann 
@Olina-Zhang 
@merriemcgaw 
Founded issues  fixed now

https://github.com/user-attachments/assets/28d036c2-7320-4d82-a880-3fad7212fa6a
